main.cpp

```diff
@@ -13,75 +13,606 @@ What you create in this project will be used as the basis of Project 5 in the co
+
+    void mixSignals(AudioMixer Digico,
+                    std::string application = "Touring Live Sound");
+    void splitSignal(std::string destination = "Broadcast");
+    void processSignal(AudioMixer AllenHeath,
+                       int paramEQ = 64,
+                       int graphEQ = 16,
+                       int expansionPorts = 8);
+};
+
+/*
+Thing 5) ADSR
+5 properties:
+    1) attack time (float)
+    2) hold time (float)
+    3) decay time (float)
+    4) sustain level (float)
+    5) release time (float)
+3 things it can do:
+    1) modify the loudness of a sound 
+    2) modify the oscillator pitch
+    3) modify the filter frequency
+ */
+
+struct ADSR
+{
+    //attack time 
+    float attackTime = 0.01f;
+    //hold time
+    float holdTime = 0.001f;
+    //decay time
+    float decayTime = 0.05f;
+    //sustain level
+    float sustainLevel = 0.5f;
+    //release time
+    float releaseTime = 0.5f;
+
+    //modify the loudness of a sound 
+    void modifyLoudness(float attackTime, float sustainLevel);
+    //modify the oscillator pitch
+    void modOscillatorPitch(float oscFreq = 440.0f,float pitch = 0.35f,float pitchMod = 0.35f);
+    //modify the filter frequency
+    void modFilterFrequency(int cutoff = 2'500, float cutoffMod = 0.35f);
+};
+
+/*
+Thing 6) LFO
+5 properties:
+    1) rate (float)
+    2) phase offset (int)
+    3) amount (float)
+    4) waveform shape (std::string) 
+    5) bypass state (bool) 
+3 things it can do:
+    1) modulate an audio signal
+    2) toggle ON and OFF
+    3) change the interaction time between signals
+ */
+
+struct LFO
+{
+    //rate
+    float rate = 20.0f; 
+    //phase offset [Hz]
+    int phaseOffset  = 0.0f; 
+    //amount [Semitones] 
+    float amount = 48; 
+    //waveform shape 
+    std::string shapeWaveform  = "Sine"; 
+    //bypass state
+    bool bypassState = false; 
+
+    void modulateSignal(std::string routeAssign = "Oscillator");
+    void toggleONOFF(bool LFO = true);
+    void signalsInteraction(float amount, int phaseOffset);
+};
+
+/*
+Thing 7) oscillator
+5 properties:
+    1) frequency (float)
+    2) finetune (float)
+    3) waveform shape (std::string)  
+    4) pulse width (float) 
+    5) octave (int)  
+3 things it can do:
+    1) generate audio signals
+    2) load samples from ROM
+    3) playback samples from ROM
+ */
+
+struct Oscillator
+{
+    //frequency [Hz]
+    float frequency = 440; 
+    //Frequency fine tune variation
+    float finetune = 0.01f; 
+    //waveform shape (Signal shape)
+    std::string waveformShape = "Square"; 
+    //pulse width (% of the positive/negative signal amplitude to be wider or smaller )
+    float pulseWidth = 0.3f; 
+    //octave (n times the original frequency )
+    int octave = 2; 
+
+    //generate audio signals
+    void generateSignal(Oscillator Square);
+    //load samples from ROM
+    void loadROMSamples(std::string selectStorageDevice = "SD",                        bool isAudioFormat = true);
```

Comment:
Fix this formatting. There is a huge gap between function parameters.
- [ ] revise